hydra/core/plugins.py

```diff
@@ -195,6 +195,7 @@ def _scan_all_plugins(
                                 inspect.isclass(obj)
                                 and issubclass(obj, Plugin)
                                 and not inspect.isabstract(obj)
+                                and obj.__module__ == loaded_mod.__name__
```

Comment:
I found another option in order to prevent those kind of issues. Here I check if the loaded object is implemented in the respective module, which does prevent adding plugins multiple times simply by importing the respective class somewhere.

Reply:
I like this, but let's do it in a separate PR, and with tests, okay?

Reply:
Sure! Updated the PR with the solution using on demand imports.